### PR TITLE
fix: comply with ivy-core sprint 17 interfaces of 'project'

### PR DESCRIPTION
**Pull request by ivy-rew**

### CODE REVIEW

File: dmn-decision-table/src/com/axonivy/ivy/process/element/rule/ui/DecisionTableEditor.java [hunk 1]
```diff
@@ -226,9 +226,10 @@ private static IVariable[] getSampleScriptContext() {
     return new IVariable[] {};
   }
 
-  public void setProject(IIvyProject ivyProject) {
-    scriptEngine = ivyProject.getIvyScriptEngine();
-    project = ivyProject.project();
+  public void setProject(Project project) {
+    this.project = project;
+    var data = IDataClassManager.instance().getProjectDataModelFor(project);
```
Comment (ivy-cst) [inline dmn-decision-table/src/com/axonivy/ivy/process/element/rule/ui/DecisionTableEditor.java:231]: Faster way would be: `IProjectDataClassManager.of(project)`
Or even: `IvyScriptEngineMapper.from(project)`
But there we have to align us first with the naming!

Comment (ivy-rew) [reply dmn-decision-table/src/com/axonivy/ivy/process/element/rule/ui/DecisionTableEditor.java:231]: ahh nice, thanks for the hint.

... to me 'Mapper' suffix is hard to get used to. I'm always searching for "Adapters" first :relaxed:

Comment (ivy-cst) [reply dmn-decision-table/src/com/axonivy/ivy/process/element/rule/ui/DecisionTableEditor.java:231]: Yes may be, but Adapters are mostly empty impls in our context by now ....

Comment (ivy-cst) [reply dmn-decision-table/src/com/axonivy/ivy/process/element/rule/ui/DecisionTableEditor.java:231]: Other idea would be just `ToIvyScriptEngine.from(project)`

**Pull request merged**